rclcpp/src/rclcpp/executor.cpp

```diff
@@ -331,6 +331,23 @@ Executor::execute_subscription(
       rcl_reset_error();
     }
     subscription->return_serialized_message(serialized_msg);
+  } else if (subscription->can_loan_messages()) {
+    auto loaned_msg = subscription->loan_message();
+    auto ret = rcl_take_loaned_message(
```

Comment:
Shouldn't think return a loaned message? Why are we getting one from the subscription first?

Reply:
I opted for an _in-out_ parameter to have a return value indicating if any error occurred.
Would you recommend just returning `nullptr` if error and check accordingly?

Reply:
It's not an in-out parameter is it? If so why do you initialize it with `subscription->loan_message()`?